client/app/components/packages/save-button.hbs

```diff
@@ -0,0 +1,17 @@
+<button
+  type="button"
+  class="button expanded large {{if @isSaved 'button--saved'}}"
+  disabled={{not @isEnabled}}
+  {{on "click" @clickEvent}}
+  ...attributes
+>
+  {{#if @isRunning}}
+    <FaIcon @icon="spinner" @spin={{true}} @pulse={{true}} class="small-margin-right" />
+    {{if @isSubmit 'Submitting' 'Saving'}}
+  {{else if @isSaved}}
+    <FaIcon @icon="check" class="small-margin-right" />
+    {{if @isSubmit 'Submitted' 'Saved'}}
+  {{else}}
+    {{if @isSubmit 'Submit' 'Save'}}
+  {{/if}}
```

Comment:
Perfect.

Only way I can think to make this better is with named blocks — then the consumer would have more control over the naming of the state, and then this component wouldn't care about the specifics of the language:

```handlebars
<SaveButton as |state|>
  <state.running>
    Saving
  </state.running>
  <state.isSaved>
    Saved
  </state.isSaved>
  <state.isIdle>
     Save
  </state.isIdle>
</SaveButton>
```

The RFC has been accepted for a while: https://github.com/emberjs/rfcs/blob/master/text/0226-named-blocks.md.  Jut a matter of time!

Anyway, this is perf.